timemachine/fe/mcgregor.py

```diff
@@ -468,6 +471,9 @@ def recursion(
     # leaf-node, every atom has been mapped
     if layer == n_a:
         if num_edges == threshold:
+            if not leaf_filter_fxn(atom_map_1_to_2):
```

Comment:
Now that we have performance implications for the number of leaf nodes we visit, should we keep track of that in the MCSDiagonstics object?

Reply:
Added `MCSDiagnostics.total_leaves_visited` in https://github.com/proteneer/timemachine/pull/1356/commits/597181663262f5780b10b06bf5b5f76c66805db2